packages/dashboards/tests/acceptance/dashboards-test.js

```diff
@@ -395,6 +405,38 @@ module('Acceptance | Dashboards', function(hooks) {
       ['Mobile DAU Goal', 'Mobile DAU Graph', 'Mobile DAU Table', 'Untitled Widget'],
       '"Untitled Widget" has been added to dashboard'
     );
+
+    // hover css events are hard
+    find('.navi-widget__actions').style.visibility = 'visible';
+    await click('.navi-widget__explore-btn');
+
+    assert.equal(currentURL(), '/dashboards/1/widgets/1/view', 'Taken to explore widget page');
+
+    await click(findAll('.navi-report-widget__breadcrumb-link')[1]);
+
+    assert.equal(currentURL(), '/dashboards/1/view', 'Taken back to dashboard page');
+
+    assert.deepEqual(
+      widgetsAfter,
+      ['Mobile DAU Goal', 'Mobile DAU Graph', 'Mobile DAU Table', 'Untitled Widget'],
+      '"Untitled Widget" is still on dashboard after navigating to subroute'
+    );
+
+    window.confirm = () => {
```

Comment:
Do we have a test for rolling back the dashboard's unsaved changes through this confirmation window?

Reply:
it's covered in a dashboard-filters test